File: leetcode/BinarySearch/leetcode-153/153.go
package binarysearch

func findMin(nums []int) int {
	if len(nums) < 2 {
		return nums[0]
	}

	left, right := 0, len(nums)-1

	for left < right-1 {
		mid := (right-left)/2 + left
		if nums[mid] < nums[right] {
			right = mid
		} else {
			left = mid
		}
	}

	if nums[left] < nums[right] {
		return nums[left]
	}
	return nums[right]
}

File: leetcode/BinarySearch/leetcode-153/153.java
class Solution {
    public int findMin(int[] nums) {
        if (nums.length == 1) {
            return nums[0];
        }

        int left = 0, right = nums.length - 1;
        while (left < right - 1) {
            int mid = (right - left) / 2 + left;
            if (nums[mid] < nums[right]) {
                right = mid;
            } else {
                left = mid;
            }
        }
        return nums[left] < nums[right] ? nums[left] : nums[right];
    }
}